.github/workflows/overview-dag.yml

```
@@ -0,0 +1,140 @@
+        if: github.event_name == 'pull_request'
+        run: |
+          cp -f docs/overview-dag.svg "$RUNNER_TEMP/overview-dag.svg"
+          cp -f docs/overview-dag.dot "$RUNNER_TEMP/overview-dag.dot"
+      - name: Verify DAG outputs
+        run: |
+          test -f docs/overview-dag.dot || { echo "docs/overview-dag.dot missing" >&2; exit 1; }
+          test -f docs/overview-dag.svg || { echo "docs/overview-dag.svg missing" >&2; exit 1; }
+          # Sanity check SVG begins with XML or <svg>
+          (head -c 5 docs/overview-dag.svg | grep -q '<?xml' || head -n1 docs/overview-dag.svg | grep -q '^<svg') || { echo 'Invalid SVG header' >&2; head -n2 docs/overview-dag.svg; exit 1; }
+      - name: Summarize DAG artifacts
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2046:warning:2:58: Quote this to prevent word splitting

(shellcheck)